assets/js/admin/meta-boxes-coupon.js

```diff
@@ -12,6 +12,14 @@ jQuery(function( $ ) {
 			$( 'select#discount_type' )
 				.on( 'change', this.type_options )
 				.change();
+
+			// Add buttons to coupon screen
+			var $coupon_screen = $( '.post-new-php.post-type-shop_coupon' ),
```

Comment:
`'.post-new-php.post-type-shop_coupon'` works great, but we could keep this button later, so allow people to generate again when is saved.